python/pyspark/sql/pandas/serializers.py

```diff
@@ -180,7 +173,7 @@ def create_array(s, t):
                 if len(s) == 0 and len(s.columns) == 0:
                     arrs_names = [(pa.array([], type=field.type), field.name) for field in t]
                 # Assign result columns by schema name if user labeled with strings
-                elif self._assign_cols_by_name and any(isinstance(name, basestring)
+                elif self._assign_cols_by_name and any(isinstance(name, str)
```

Comment:
We might want to think about removing this as an option as a followup. It was mostly added because dataframe constructed with python < 3.6 could not guarantee the order of columns, but now it should match the given schema.

Reply:
Ah, right. sounds good!

Reply:
and yes, maybe it should better be in a separate PR.